tests/test_resource_tracker.py

```diff
@@ -38,45 +44,54 @@ def test_child_retrieves_resource_tracker(self):
         # child process. If the two processes do not share the same
         # resource_tracker, a cache KeyError should be printed in stderr.
         import subprocess
-        folder_name = 'loky_tempfolder'
         cmd = '''if 1:
         import os, sys
 
         from loky import ProcessPoolExecutor
         from loky.backend import resource_tracker
         from loky.backend.semlock import SemLock
+        from tempfile import NamedTemporaryFile
+        import sys
 
+
+        tmpfile = NamedTemporaryFile(delete=False)
+        tmpfile.close()
+        filename = tmpfile.name
         resource_tracker.VERBOSE=True
-        folder_name = "{}"
 
-        # We don't need to create the semaphore as registering / unregistering
-        # operations simply add / remove entries from a cache, but do not
-        # manipulate the actual semaphores.
-        resource_tracker.register(folder_name, "folder")
+        resource_tracker.register(filename, "file")
 
-        def unregister(name, rtype):
-            # resource_tracker.unregister is actually a bound method of the
+        def maybe_unlink(name, rtype):
+            # resource_tracker.maybe_unlink is actually a bound method of the
             # ResourceTracker. We need a custom wrapper to avoid object
             # serialization.
```

Comment:
I don't understand this: this turns a bound method into a local closure that references the same `resource_tracker` object. In both cases the `resource_tracker` object would be pickled, no?

Reply:
Nope: `cloudpickle` only pickles objects defined in outer scopes. Here, `resource_tracker` is a local, and thus not pickled. It will be re-imported through the byte-code execution of the  `maybe_unlink` function.

Reply:
Alright.